public/js/utils/source-map.js

```diff
@@ -7,11 +7,16 @@ const { getSource } = require("../selectors");
 const { isEnabled } = require("../feature");
 const { isJavaScript } = require("./source");
 const assert = require("./assert");
+const { prefs } = require("./prefs");
 
 import type { Location } from "./actions/types";
 
 let sourceMapRequests = new Map();
 
+function shouldSourceMap() {
+  return isEnabled("sourceMaps") && prefs.sourceMapsEnabled;
```

Comment:
Thanks for bringing this up. We've talked about incorporating Prefs for a couple of months and it's nice that we're at the point where we can act on it.

I don't think this issue needs to block on re-hauling our config system. We've seen in the past that the config system is surprisingly gnarly and I want to take the time to think it through.

I like the idea of reading Prefs at startup and using `isEnabled` and `getValue`.

We will likely need to keep JSON configs as some of the config values are used for the node scripts.